Refactor automatic scaling implementation for NonlinearEigenSystem

lindsayad opened the pull request:
Moved `computeScaling` from `NonlinearSystem` down into
`NonlinearSystemBase` so that all derived classes may make use of the
method.

Closes #15048

lindsayad committed:
Refactor automatic scaling implementation for NonlinearEigenSystem

Moved `computeScaling` from `NonlinearSystem` down into
`NonlinearSystemBase` so that all derived class may make use of the
method.

Closes #15048

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/557049/) on 88f20ff wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15531/site)

This comment will be updated on new commits.

lindsayad committed:
Add eigen scaling tests

Test scaling based on both the preconditioning matrix and on the
non-eigen residual vector

Refs #15048

fdkong commented on test/tests/problems/eigen_problem/eigensolvers/ne-coupled-resid-scaling.i:
```diff
@@ -0,0 +1,136 @@
+  [./diff_T]
+    type = CoefDiffusion
+    variable = T
+    coef = 1e30
```
Oh, beautiful number. How about 1e3000?

lindsayad replied:
Maybe we can overflow the input file

fdkong commented:
It looks nice to me.

fdkong commented:
@YaqiWang Do you have comments?

YaqiWang commented:
Sorry, did not notice this until I check recently merged PRs in MOOSE. Should be good. Glad to see you guys working on these issues!
